src/Stratis.Bitcoin.Features.Consensus/ConsensusLoop.cs

```diff
@@ -395,16 +397,22 @@ public void Stop()
                         return;
                     }
 
+
                     // Set the chain back to ConsensusLoop.Tip.
                     this.Chain.SetTip(this.Tip);
                     this.logger.LogTrace("Chain reverted back to block '{0}'.", this.Tip);
-
-                    if ((blockValidationContext.Peer != null) && (blockValidationContext.BanDurationSeconds != BlockValidationContext.BanDurationNoBan))
+    
+                    if (blockValidationContext.Peer != null && blockValidationContext.CanBanPeers)
                     {
                         int banDuration = blockValidationContext.BanDurationSeconds == BlockValidationContext.BanDurationDefaultBan ? this.connectionManager.ConnectionSettings.BanTimeSeconds : blockValidationContext.BanDurationSeconds;
                         this.peerBanning.BanPeer(blockValidationContext.Peer, banDuration, $"Invalid block received: {blockValidationContext.Error.Message}");
                     }
 
+                    if (blockValidationContext.Error == ConsensusErrors.BadTransactionDuplicate)
+                    {
+                        return;
```

Comment:
Can you please add exit log here?

Reply:
Can I ask you why ?
I honestly think there is already way too much nesting in this method, and I don't want to do more unless it is necessary.

Reply:
Please read https://github.com/stratisproject/StratisBitcoinFullNode/blob/master/Documentation/logging-style.md. You created exit point from the function without proper logging it. The most elegant solution here is to invert the condition and then just follow through to the function end, which contains the log. Alternative solution would be as you have it but with correct logging and that would also be okay.